### ISSUE TITLE
oxlint-tailwindcss does not generate optimal class names for arbitrary values

### ISSUE THREAD

**Issue by HM-Suiji** [sergioazoc/oxlint-tailwindcss#15]
## Description:
When using the TailwindCSS VSCode plugin, it sometimes suggests that classes can be written in a more canonical form. For example:

The class `border-(--border)` can be written as `border-border` (suggestCanonicalClasses)

However, the oxlint-tailwindcss plugin, while correctly formatting arbitrary value syntax like border-[var(--border)], does not convert it to the recommended canonical form border-border.

## Expected Behavior:
It would be ideal if oxlint-tailwindcss could respect Tailwind’s canonical class suggestions when formatting arbitrary values, outputting the most optimal class name rather than the raw variable syntax.

## Screenshots:

<img width="1671" height="641" alt="Image" src="https://github.com/user-attachments/assets/ca66d176-2b0b-4a5e-8d4d-336c4c4372a9" />

## Environment:

VSCode TailwindCSS extension version: 0.14.29
oxlint-tailwindcss version: 0.6.3
Node.js version: 25.3.0

**Comment by sergioazoc** [comment]
Shipping in **v0.7.0** as a new opt-in rule: `tailwindcss/prefer-theme-tokens` ([3752776](https://github.com/sergioazoc/oxlint-tailwindcss/commit/3752776)). Thanks for the report.

**Quick context on the design.** `enforce-canonical` already calls Tailwind's `canonicalizeCandidates()` API, which *does* resolve theme-token references when the variable name is the standard one — for example `border-(--color-border)` → `border-border` and `border-[var(--color-border)]` → `border-border`. So the simple case was already covered.

The case in your screenshot (`border-(--border)` → `border-border`) is different: `--border` is the *raw* CSS variable, and the actual Tailwind theme token is `--color-border` (often defined as `hsl(var(--border))` in shadcn-style themes). `border-(--border)` and `border-border` produce **different CSS** in those setups (`var(--border)` vs `hsl(var(--border))`), so `canonicalizeCandidates` correctly leaves them alone.

The VS Code Tailwind extension's `suggestCanonicalClasses` uses a softer heuristic — name match: if `prefix-(--name)` exists and `prefix-name` is a valid utility, suggest the replacement, regardless of whether the CSS is byte-identical. That's the heuristic this new rule implements:

```jsonc
// .oxlintrc.json
"rules": {
  "tailwindcss/prefer-theme-tokens": "warn"
}
```

```tsx
// ❌ Bad → ✅ Fixed (when border-border is a valid utility)
"border-(--border)"        → "border-border"
"bg-[var(--primary)]"      → "bg-primary"
"hover:bg-(--primary)/80"  → "hover:bg-primary/80"
```

**Off by default** because the replacement may change observable CSS in shadcn-style themes (e.g. `var(--border)` is not a valid color value on its own; `border-border` resolves to `hsl(var(--border))`). It coexists with `enforce-canonical` and `no-unnecessary-arbitrary-value` without double-firing — boundary verified by an integration matrix test.

**Comment by HM-Suiji** [comment]
> Shipping in **v0.7.0** as a new opt-in rule: `tailwindcss/prefer-theme-tokens` ([3752776](https://github.com/sergioazoc/oxlint-tailwindcss/commit/3752776)). Thanks for the report.
> 
> **Quick context on the design.** `enforce-canonical` already calls Tailwind's `canonicalizeCandidates()` API, which _does_ resolve theme-token references when the variable name is the standard one — for example `border-(--color-border)` → `border-border` and `border-[var(--color-border)]` → `border-border`. So the simple case was already covered.
> 
> The case in your screenshot (`border-(--border)` → `border-border`) is different: `--border` is the _raw_ CSS variable, and the actual Tailwind theme token is `--color-border` (often defined as `hsl(var(--border))` in shadcn-style themes). `border-(--border)` and `border-border` produce **different CSS** in those setups (`var(--border)` vs `hsl(var(--border))`), so `canonicalizeCandidates` correctly leaves them alone.
> 
> The VS Code Tailwind extension's `suggestCanonicalClasses` uses a softer heuristic — name match: if `prefix-(--name)` exists and `prefix-name` is a valid utility, suggest the replacement, regardless of whether the CSS is byte-identical. That's the heuristic this new rule implements:
> 
> // .oxlintrc.json
> "rules": {
>   "tailwindcss/prefer-theme-tokens": "warn"
> }
> // ❌ Bad → ✅ Fixed (when border-border is a valid utility)
> "border-(--border)"        → "border-border"
> "bg-[var(--primary)]"      → "bg-primary"
> "hover:bg-(--primary)/80"  → "hover:bg-primary/80"
> **Off by default** because the replacement may change observable CSS in shadcn-style themes (e.g. `var(--border)` is not a valid color value on its own; `border-border` resolves to `hsl(var(--border))`). It coexists with `enforce-canonical` and `no-unnecessary-arbitrary-value` without double-firing — boundary verified by an integration matrix test.

Thank you for your explanation. I'm not using shadcn, so this rule set is very useful to me.
